File: src/main/java/com/example/dalattravel/model/LegacyLocation.java
package com.example.DaLattravel.model;

import jakarta.persistence.*;
import lombok.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "legacy_locations", indexes = {
    @Index(name = "IX_LegacyLocation_GPS", columnList = "latitude, longitude"),
    @Index(name = "IX_LegacyLocation_IsActive", columnList = "isActive"),
    @Index(name = "IX_LegacyLocation_IsMerged", columnList = "isMergedLocation")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LegacyLocation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 100)
    private String currentName;

    @Column(length = 100)
    private String oldName;

    private boolean isMergedLocation;

    @Column(columnDefinition = "TEXT")
    private String mergeNote;

    private double latitude;
    private double longitude;

    @Builder.Default
    private boolean isActive = true;

    @OneToMany(mappedBy = "legacyLocation", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @Builder.Default
    private List<TransportPriceHistory> priceHistories = new ArrayList<>();
}

File: DoAnChuyenNganh-main/DoAnCoSo/WebDuLichDaLat/Models/LegacyLocation.cs
using System.ComponentModel.DataAnnotations;

namespace WebDuLichDaLat.Models
{
    public class LegacyLocation
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string OldName { get; set; }  // Ví dụ: "Long An"

        [Required]
        [StringLength(100)]
        public string CurrentName { get; set; } // Ví dụ: "Tây Ninh"

        public double Latitude { get; set; }
        public double Longitude { get; set; }

        //  THÊM MỚI: Thông tin sáp nhập tỉnh thành
        public bool IsMergedLocation { get; set; }     // true nếu là tỉnh bị sáp nhập
        
        public DateTime? MergeDate { get; set; }       // Ngày sáp nhập (01/07/2025)
        
        [StringLength(255)]
        public string? MergeNote { get; set; }         // Ghi chú về sáp nhập
        
        public bool IsActive { get; set; } = true;     // false nếu không còn dùng

        // Quan hệ ngược
        public virtual ICollection<TransportPriceHistory> PriceHistories { get; set; } = new List<TransportPriceHistory>();
    }
}

